src/ui/common/src/components/integrations/addIntegrations.tsx

```diff
@@ -148,20 +162,231 @@ const AddIntegrationListItem: React.FC<AddIntegrationListItemProps> = ({
     </Box>
   );
 
+  if (service !== 'Kubernetes') {
+    return (
+      <Box key={service}>
+        <Box>
+          {iconWrapper}
+          {showDialog && (
+            <IntegrationDialog
+              user={user}
+              service={service}
+              onSuccess={() => {
+                setShowDialog(false);
+                setShowSuccessToast(service);
+              }}
+              onCloseDialog={() => {
+                setShowDialog(false);
+                dispatch(resetConnectNewStatus());
+              }}
+              showMigrationDialog={() => setShowMigrationDialog(true)}
+            />
+          )}
+        </Box>
+        <Snackbar
+          anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
+          open={showSuccessToast === service}
+          onClose={handleSuccessToastClose}
+          key={'integrations-dialog-success-snackbar'}
+          autoHideDuration={6000}
+        >
+          <Alert
+            onClose={handleSuccessToastClose}
+            severity="success"
+            sx={{ width: '100%' }}
+          >
+            {`Successfully connected to ${service}!`}
+          </Alert>
+        </Snackbar>
+      </Box>
+    );
+  }
+
+  const handleRegularK8s = () => {
+    setShowKubernetesDialog(true);
+    setShowDialog(false);
+  };
+
+  const handleOndemandK8s = () => {
+    setShowSelectProviderDialog(true);
+    setShowDialog(false);
+  };
+
+  const handlePrevious = () => {
+    setShowSelectProviderDialog(false);
+    setShowDialog(true);
+  };
+
+  const handleAWSClick = () => {
+    setShowOndemandDialog(true);
+    setShowSelectProviderDialog(false);
+  };
+
   return (
     <Box key={service}>
       <Box>
         {iconWrapper}
-        {showDialog && (
+        <Dialog open={showDialog} onClose={() => setShowDialog(false)}>
```

Comment:
Same here, this is very Kubernetes specific and has nothing to do with the other integrations. This should have it's own file somewhere.